tests/integration/test_solve.py

```diff
@@ -80,6 +80,109 @@ def test_solve_corner_cases():
     assert allclose(b, num.matmul(a, out))
 
 
+def test_solve_b_is_empty():
+    a = num.random.rand(1, 1)
+    b = num.atleast_2d([])
+
+    out = num.linalg.solve(a, b)
+    assert np.array_equal(b, out)
+
+
+def test_solve_dtype_int():
+    n = 8
+    a = np.random.randint(0, 10, size=(n, n), dtype="i4")
+    b = np.random.randint(0, 10, size=(n,), dtype="i4")
```

Comment:
do these not work with 64 bit int types?

Reply:
BTW I do believe that there is a non-vanishing probability that a random matrix of non-negative integers is singular. Is there a reason to make these random?

Reply:
I test both np.int32 and np.int64 now. 
You are right. There is probability that the random int matrix is singular. Fixed.